src/backend/utils/storage.js

```diff
@@ -1,16 +1,38 @@
+const crypto = require('crypto');
+const normalizeUrl = require('normalize-url');
+const { logger } = require('./logger');
 const { redis } = require('../lib/redis');
 
+const standardize = (url, type) => {
```

Comment:
Can we rename this `createKey(uri, prefix)`?  We will either pass a URL or a URI, so URI makes sense here, and the "type" is really just a string prefix in our key's namespace.

Reply:
Renamed to `createKey(uri, prefix)`.